www/src/utils/typography.js

```diff
@@ -108,102 +108,95 @@ const _options = {
       ".gatsby-highlight": {
         background: colors.code.bg,
         borderRadius: `${presets.radius}px`,
-        padding: rhythm(options.blockMarginBottom),
-        marginBottom: rhythm(options.blockMarginBottom),
-        overflow: `auto`,
-        WebkitOverflowScrolling: `touch`,
         position: `relative`,
       },
       ".gatsby-highlight pre[class*='language-']": {
-        padding: 0,
-        marginTop: 0,
-        marginBottom: 0,
         backgroundColor: `transparent`,
         border: 0,
-        float: `left`,
-        minWidth: `100%`,
-        overflow: `initial`,
+        WebkitOverflowScrolling: `touch`,
       },
-      ".gatsby-highlight pre[class*='language-']::before": {
+      ".gatsby-highlight::before": {
+        fontFamily: `monospace`,
         position: `absolute`,
         top: `0px`,
         right: `20px`,
         padding: `3px 10px`,
         fontSize: `12px`,
-        textAlign: `right`,
         color: `#444`,
         fontWeight: `700`,
         letterSpacing: `0.8px`,
         textTransform: `uppercase`,
         borderRadius: `0 0 5px 5px`,
         background: `#ddd`,
       },
-      ".gatsby-highlight pre[class='language-javascript']::before": {
+      ".gatsby-highlight[data-language='javascript']::before": {
```

Comment:
@DSchau After this change, the type tab will disappear if the `data-language` is not included.

Reply:
cc @amberleyromo is this behavior we'd even want?

@Raincal this makes this functionality opt-in correct? I think we want this on all of our language blocks.

Reply:
In .md files, the data-language attribute will be added by gatsby-remark-prismjs.
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-prismjs/src/index.js

Reply:
Right - but is there a technical reason to make it opt-in? We already sort of have that functionality if we wanted it, because we (granularly) would just remove the style for that type of code block we don't want highlighted.

In other words--just trying to understand why the change in behavior was made! Thanks for understanding!

Reply:
@DSchau You are right. It's not necessary to use `data-language`.

Reply:
@Raincal Still catching up on things (unfortunately after we merged :/).
I just checked the codesandbox demo — thanks for putting that together! 🙏 

So to summarize:

- this PR makes `pre` the container that provides the scrollbar for overflowing content (as opposed to `.gatsby-highlight`)
- Mobile Safari doesn't respect `position: absolute` in overflowing content, so if we attach the language badges to its `before` pseudo element, they will scroll with the content on Mobile Safari (see @Raincal's demo)
- this is why @Raincal's original PR attached the badge to `.gatsby-highlight` instead, but that change was reverted in a453b22 after the thread here

@DSchau Looking at https://github.com/gatsbyjs/gatsby/blob/794fb4be4c47744379d575f68021987bec4d74fb/packages/gatsby-remark-prismjs/src/index.js#L69-L78 and also https://github.com/gatsbyjs/gatsby/blob/794fb4be4c47744379d575f68021987bec4d74fb/packages/gatsby-remark-prismjs/src/index.js#L39-L42, I'm thinking we should always have the same values for `data-language` on `.gatsby-highlight` and `class` on `.gatsby-hightlight pre`, and a default value of `text` … so I think @Raincal's original approach would not have caused any change irt to "opt-in/out"?

Maybe I'm not seeing the forest before the trees, but I think the change suggested by @Raincal should not alter code badges in any way other than fixing the "scrolling away" problem not only for desktop browsers, but also for Mobile Safari.

@Raincal Would love if you wanted to chime in if you find the time!